services/managers/discourse_manager.py

```diff
@@ -0,0 +1,338 @@
+                return g['name']
+        raise KeyError("Group ID %s not found on Discourse" % id)
+
+    @staticmethod
+    def __add_user_to_group(id, username):
+        endpoint = ENDPOINTS['groups']['add_user']
+        DiscourseManager.__exc(endpoint, id, usernames=[username])
+
+    @staticmethod
+    def __remove_user_from_group(id, username):
+        endpoint = ENDPOINTS['groups']['remove_user']
+        DiscourseManager.__exc(endpoint, id, username=username)
+
+    @staticmethod
+    def __generate_group_dict(names):
+        dict = {}
+        for name in names:
+            dict[name] = DiscourseManager.__group_name_to_id(name)
+        return dict
+
+    @staticmethod
+    def __get_user_groups(username):
+        data = DiscourseManager.__get_user(username)
+        return [g['id'] for g in data['user']['groups'] if not g['automatic']]
+
+    @staticmethod
+    def __user_name_to_id(name):
+        data = DiscourseManager.__get_user(name)
+        return data['user']['id']
+
+    @staticmethod
+    def __user_id_to_name(id):
+        raise NotImplementedError
+
+    @staticmethod
+    def __get_user(username):
+        endpoint = ENDPOINTS['users']['get']
+        return DiscourseManager.__exc(endpoint, username)
+
+    @staticmethod
+    def __activate_user(username):
+        endpoint = ENDPOINTS['users']['activate']
+        id = DiscourseManager.__user_name_to_id(username)
+        DiscourseManager.__exc(endpoint, id)
+
+    @staticmethod
+    def __update_user(username, **kwargs):
+        endpoint = ENDPOINTS['users']['update']
+        id = DiscourseManager.__user_name_to_id(username)
+        DiscourseManager.__exc(endpoint, id, params=kwargs)
+
+    @staticmethod
+    def __create_user(username, email, password):
+        endpoint = ENDPOINTS['users']['create']
+        DiscourseManager.__exc(endpoint, name=username, username=username, email=email, password=password, active=True)
+
+    @staticmethod
+    def __check_if_user_exists(username):
+        try:
+            DiscourseManager.__user_name_to_id(username)
+            return True
+        except:
+            return False
+
+    @staticmethod
+    def __suspend_user(username):
+        id = DiscourseManager.__user_name_to_id(username)
+        endpoint = ENDPOINTS['users']['suspend']
+        return DiscourseManager.__exc(endpoint, id, duration=DiscourseManager.SUSPEND_DAYS, reason=DiscourseManager.SUSPEND_REASON)
+
+    @staticmethod
+    def __unsuspend(username):
+        id = DiscourseManager.__user_name_to_id(username)
+        endpoint = ENDPOINTS['users']['unsuspend']
+        return DiscourseManager.__exc(endpoint, id)
+
+    @staticmethod
+    def __set_email(username, email):
+        endpoint = ENDPOINTS['users']['set_email']
+        return DiscourseManager.__exc(endpoint, username, email=email)
+
+    @staticmethod
+    def _sanatize_username(username):
+        sanatized = username.replace(" ", "_")
+        sanatized = sanatized.replace("'", "")
```

Comment:
well I was only trying to help you out. but thats fine I wont "nit pick" anymore